src/Proto.Actor/Metrics/Metrics.cs

```diff
@@ -1,42 +1,17 @@
-using System.Linq;
+using System.Diagnostics.Metrics;
 using JetBrains.Annotations;
-using Proto.Utils;
-using Ubiquitous.Metrics;
-using Ubiquitous.Metrics.Labels;
-using Ubiquitous.Metrics.NoMetrics;
 
 namespace Proto.Metrics
 {
     [PublicAPI]
     public class ProtoMetrics
     {
-        internal readonly ActorMetrics InternalActorMetrics;
-        public readonly bool IsNoop;
-        private IMetricsProvider[] _configurators;
-        private TypeDictionary<object,ProtoMetrics> _knownMetrics = new(10);
-        private Ubiquitous.Metrics.Metrics _metrics;
-
-        public ProtoMetrics(IMetricsProvider[] configurators)
-        {
-            if (configurators.FirstOrDefault() is NoMetricsProvider) IsNoop = true;
-
-            _metrics = Ubiquitous.Metrics.Metrics.CreateUsing(configurators);
-            _configurators = configurators;
-            InternalActorMetrics = new ActorMetrics(this);
-            Register(InternalActorMetrics);
-        }
+        public const string MeterName = "Proto.Actor";
 
-        public void Register<T>(T instance) => _knownMetrics.Add<T>(instance!);
+        public static Meter Meter = new(MeterName, typeof(ProtoMetrics).Assembly.GetName().Version?.ToString());
 
-        public T Get<T>() => (T) _knownMetrics.Get<T>()!;
-
-        public ICountMetric CreateCount(string name, string description, params LabelName[] labelNames)
-            => _metrics.CreateCount(name, description, labelNames);
-
-        public IGaugeMetric CreateGauge(string name, string description, params LabelName[] labelNames)
-            => _metrics.CreateGauge(name, description, labelNames);
+        public readonly bool IsNoop;
 
-        public IHistogramMetric CreateHistogram(string name, string description, params LabelName[] labelNames)
-            => _metrics.CreateHistogram(name, description, labelNames);
+        public ProtoMetrics(bool recordMetrics) => IsNoop = !recordMetrics;
```

Comment:
.Enabled? It is almost exclusively used with !, that could perhaps read better if it was not defined as a negative

Reply:
I'd say more, if there's no listener for the metric, it won't do anything. So, I don't really think checking if metrics are enabled or not is overkill as the `Meter` thing does it anyway. Might be useful to measure it anyway before making a decision to skip this check or not.

Reply:
Ok, it's not _exactly_ like that, but metrics report to their subscribers, so when there are no subscribers, there's still some overhead, but it's minimal. The question is if we encourage people to measure their system, checking if metrics are enabled is pointless.

Reply:
I kept the IsNoop flag checking from original implementation assuming this is a micro-optimization. If someone is not interested in metrics, they don't have to pay the price of allocations and method calls related to that. Bear in mind that many of the metrics are on the hot path, e.g. measuring the actor's Receive handler. There are many places in proto.actor code that do micro-optimizations, like [here](https://github.com/asynkron/protoactor-dotnet/blob/620e61ddc739b6d34ddb127481169c9d76437d28/src/Proto.Actor/Context/ActorContext.cs#L432)

I can change the name to Enabled though.